packages/utils/src/lib/semver.ts

```diff
@@ -0,0 +1,26 @@
+import { compare, validate } from 'compare-versions';
+
+export function normalizeSemver(semverString: string): string {
+  if (semverString.startsWith('v') || semverString.startsWith('V')) {
+    return semverString.slice(1);
+  }
+
+  if (semverString.includes('@')) {
+    return semverString.split('@').at(-1) ?? '';
+  }
+
+  return semverString;
+}
+
+export function isSemver(semverString = ''): boolean {
+  return validate(normalizeSemver(semverString));
+}
+// eslint-disable functional/immutable-data
+export function sortSemvers(semverStrings: string[]): string[] {
+  return [...semverStrings]
+    .filter(Boolean)
+    .filter(isSemver)
+    .sort((a, b) =>
+      compare(normalizeSemver(a), normalizeSemver(b), '>') ? -1 : 0,
+    );
+}
```

Comment:
There's no mutation:

```suggestion

export function sortSemvers(semverStrings: string[]): string[] {
  return semverStrings
    .filter(Boolean)
    .filter(isSemver)
    .sort((a, b) =>
      compare(normalizeSemver(a), normalizeSemver(b), '>') ? -1 : 0,
    );
}
```

Reply:
I know! for some reason the rule failed... will try again!